homework/src/index.js

```diff
@@ -31,17 +32,151 @@
+      'html': 'HYF Repositories'
+    });
+    const select = createAndAppend('select', header, {
+      'id': 'list'
+    });
+
+
+    const repositoriesInfoSec = createAndAppend('section', container, {
+      'class': 'repos-info-sec box'
+    });
+
+    const table = createAndAppend('table', repositoriesInfoSec, {
+      'id': 'info-table'
+    });
+    const repoNameLine = createAndAppend('tr', table);
+    createAndAppend('td', repoNameLine, {
+      'html': 'Repository :',
+      'class': 'label'
+    });
+    const repoName = createAndAppend('td', repoNameLine, {
+      'id': 'repoName'
+    });
+    const repoNameLink = createAndAppend('a', repoName, {
+      'target': '_blank',
+    });
+
+    const descriptionLine = createAndAppend('tr', table);
+    const descriptionLabel = createAndAppend('td', descriptionLine, {
+      'html': 'Description :',
+      'class': 'label'
+    });
+    const descriptionValue = createAndAppend('td', descriptionLine);
+    const forkLine = createAndAppend('tr', table);
+    createAndAppend('td', forkLine, {
+      'html': 'Forks :',
+      'class': 'label'
+    });
+    const forkValue = createAndAppend('td', forkLine);
+
+    const updateLine = createAndAppend('tr', table);
+    createAndAppend('td', updateLine, {
+      'html': 'Updated :',
+      'class': 'label'
+    });
+    const updateValue = createAndAppend('td', updateLine);
+    const contributorsSec = createAndAppend('section', container, {
+      'class': 'contributors-sec box'
+    });
+    createAndAppend('p', contributorsSec, {
+      html: 'Contributions',
+      'class': 'contributions'
+    });
+    const ul = createAndAppend('ul', contributorsSec);
+
+
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
+        createAndAppend('div', container, {
+          html: err.message,
+          class: 'alert-error'
+        });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {
+          ignorePunctuation: true
+        }));
+
+        data.forEach((element, i) => {
+          createAndAppend('option', select, {
+            'html': element.name,
+            'value': i
+          });
+        });
+
+        showRepositories(0, data);
+
+        const contributorsInfo = data[0].contributors_url;
+        fetchJSON(contributorsInfo, (err, contributorData) => {
+          showContributors(contributorData, ul);
+        });
+
+        select.addEventListener('change', (e) => {
+          const index = e.target.value;
+          showRepositories(index, data);
+
+          const contributorsInfo = data[index].contributors_url;
+          fetchJSON(contributorsInfo, (err, contributorData) => {
+            ul.innerHTML = '';
+            showContributors(contributorData, ul);
+          });
```

Comment:
This code is a repetition of the code in lines 188-121. You could either create a new function that fetches the contributors and passes them on to `showContributors()` or add the fetching part to the `showContributors()` function (probably then renamed as `fetchAndShowContributors()` like this:

```js
function fetchAndShowContributors(url, ul) {
  fetchJSON(contributorsInfo, (err, contributors) => {
    if (err) {
      //...
   } else {
     // the code that you now have in showContributors
   }
  });
}
```